climada/util/yearsets.py

```diff
@@ -0,0 +1,266 @@
+    # if correction_factor > 0.1:
+    #     tex = raw_input("Do you want to exclude small events?")
+
+
+    ais.date = u_dt.str_to_date(year_list)
+    ais.at_event = impact_per_year / correction_factor
+    ais.frequency = np.ones(n_resampled_years)*sum(nr_events)/n_resampled_years
+
+
+    return ais, sampling_vect, nr_events
+
+def resample_single_annual_event(eis, nr_resampled_years, sampling_vect=None):
+    """Sample one single annual event
+
+    INPUTS:
+        eis (impact class object): event impact set
+        nr_resampled_years (int): the target number of years the impact yearset shall
+          contain.
+        sampling_vect (array): the sampling vector
+
+    OUTPUTS:
+        impact_per_year (array): resampled impact per year (length = nr_resampled_years)
+        nr_events (array): amount of resampled events per year (length = nr_resampled_years)
+        sampling_vect (array): the sampling vector
+      """
+
+
+    impact_per_year = np.zeros(nr_resampled_years)
+    if not sampling_vect:
+        nr_input_events = len(eis.event_id)
+        sampling_vect = sampling_uniform(nr_resampled_years, nr_input_events)
+
+    for idx_event, event in enumerate(sampling_vect):
+        impact_per_year[idx_event] = eis.at_event[sampling_vect[event]]
+
+    nr_events = np.ones(nr_resampled_years)
+
+    return impact_per_year, nr_events, sampling_vect
+
+def resample_multiple_annual_events(eis, nr_resampled_years, nr_events=None, sampling_vect=None):
+    """Sample multiple events per year
+
+    INPUTS:
+        eis (impact class object): event impact set
+        nr_resampled_years (int): the target number of years the impact yearset shall
+          contain.
+        nr_annual_events (int): number of events per year in given event impact set
+        sampling_vect (array): the sampling vector
+
+    OUTPUTS:
+        impact_per_year (array): resampled impact per year (length = nr_resampled_years)
+        nr_events (array): amount of resampled events per year (length = nr_resampled_years)
+        sampling_vect (array): the sampling vector
+      """
+
+    nr_annual_events = np.sum(eis.frequency)
+
+    if not sampling_vect:
+        nr_input_events = len(eis.event_id)
+        sampling_vect, nr_events = sampling_poisson(nr_resampled_years,
+                                                    nr_annual_events, nr_input_events)
+
+    impact_per_event = np.zeros(np.sum(nr_events))
+    impact_per_year = np.zeros(nr_resampled_years)
+
+    for idx_event, event in enumerate(sampling_vect):
+        impact_per_event[idx_event] = eis.at_event[sampling_vect[event]]
+
+    idx = 0
+    for year in range(nr_resampled_years):
+        impact_per_year[year] = np.sum(impact_per_event[idx:(idx+nr_events[year])])
+        idx += nr_events[year]
+
+    return impact_per_year, nr_events, sampling_vect
+
+
+def sampling_uniform(tot_nr_events, nr_input_events):
```

Comment:
If you would use scipy, I think you can make one single sampling function for both `uniform` and `poisson`, with in addition the possibility to use any other type of discrete sampling. This would make the code significantly shorter and more general. Pleaselet me know if you think it is a good idea and I can propose an implementation.

Reply:
Yes, that sounds good! Would be great if you could propose how to implement it :).